### DISCUSSION TITLE
머테리얼을 시간마다 바꾸려고 하는데

### DISCUSSION THREAD

**Discussion by smartbh** [naverz/zepeto-studio-kor#72] [5. Q&A]
전구가 반짝반짝 거리는것과 같은 느낌으로

머테리얼을 초마다 바꾸는 코드를 짜려 했습니다.

C#기본 유니티에선 되던 방식 그대로 짜봤는데

이상하게 property를 받아 올 수가 없다 하는

오류가 뜨더라구요 다른 질문에 올린 카메라 getcomponent도 그러더니..

![image](https://user-images.githubusercontent.com/43138813/145730595-358a7722-fdc6-4ed0-a6a7-45f0a06c4a8c.png)

이런식으로 짰었는데 크게 문제가 되나요?

**Comment by Zepeto-TechEvangelist** [comment]
안녕하세요, GetComponet는 정상적으로 동작하고 있습니다.
<img width="459" alt="스크린샷 2022-03-21 오후 6 43 21" src="https://user-images.githubusercontent.com/100118650/159237036-44bb35ee-4dce-4e70-95ba-8927d3d66aae.png">

1초에 한번씩 머테리얼을 바꾸고 싶다면 코루틴방식보다 deltaTime을 쓰는 방법이 효과적입니다.
아래 코드와 위의 구조를 참고하여 주세요.

import { ZepetoScriptBehaviour } from 'ZEPETO.Script'
import * as UnityEngine from "UnityEngine";
import {Time} from "UnityEngine";

export default class ChangeMaterial extends ZepetoScriptBehaviour {
    
    public LightBulbMat1: UnityEngine.Material;
    public LightBulbMat2: UnityEngine.Material;
    rend: UnityEngine.Renderer;
    private delta: number  = 0;
    public interval: number = 1;
    private flag: bool = true;

    Start() {    
        this.rend = this.GetComponent<UnityEngine.Renderer>();
        this.rend.material = this.LightBulbMat1;
    }
    
    Update() {
        this.delta += Time.deltaTime;
        if(this.delta > this.interval)
        {
            this.delta = 0;
            this.flag = !this.flag;
        }

        if (this.flag)
        {
            //Move upward during the set moving time
            this.rend.material = this.LightBulbMat1;
        }
        else
        {
            //Move downward during the set moving time
            this.rend.material = this.LightBulbMat2;
        }
    }
}
